executor.go

```diff
@@ -714,14 +722,369 @@ func (e *executor) executeDifferenceShard(ctx context.Context, index string, c *
+	tail := product(input[1:]) // invoke product on remaining element
+	for h := range head {      // for each head
+		for t := range tail { // iterate over the tail
+			s := ppi{gCnt: GroupCount{Group: make([]FieldRow, 0)}}
+			s.gCnt.Group = append([]FieldRow{head[h].fieldRow}, tail[t].gCnt.Group...) // had to insert at the front to match input order
+			if tail[t].row != nil {                                                    // first time around nothing to intersect
+				s.row = head[h].row.Intersect(tail[t].row)
+			} else {
+				s.row = head[h].row
+			}
+			res = append(res, s)
+		}
+	}
+	return res
+}
+
+func (e *executor) executeRowIDs(ctx context.Context, index string, c *pql.Call, shards []uint64, opt *execOptions) (RowIDs, error) {
+	// Execute calls in bulk on each remote node and merge.
+	mapFn := func(shard uint64) (interface{}, error) {
+		return e.executeRowIDsShard(ctx, index, c, shard)
+	}
+	// Merge returned results at coordinating node.
+	reduceFn := func(prev, v interface{}) interface{} {
+		other, _ := prev.(RowIDs)
+		return other.merge(v.(RowIDs))
+	}
+	// Get full result set.
+	other, err := e.mapReduce(ctx, index, shards, c, opt, mapFn, reduceFn)
+	if err != nil {
+		return nil, err
+	}
+	results, _ := other.(RowIDs)
+	// Apply offset.
+	if offset, hasOffset, err := c.UintArg("offset"); err != nil {
+		return nil, err
+	} else if hasOffset {
+		if int(offset) < len(results) {
+			results = results[offset:]
+		}
+	}
+	// Apply limit.
+	if limit, hasLimit, err := c.UintArg("limit"); err != nil {
+		return nil, err
+	} else if hasLimit {
+		if int(limit) < len(results) {
+			results = results[:limit]
+		}
+	}
+	return results, nil
+}
+
+func (e *executor) executeRowIDsShard(ctx context.Context, index string, c *pql.Call, shard uint64) (RowIDs, error) {
+	// Fetch index.
+	idx := e.Holder.Index(index)
+	if idx == nil {
+		return nil, ErrIndexNotFound
+	}
+	// Fetch field name from argument.
+	fieldName, ok := c.Args["field"].(string)
+	if !ok {
+		return nil, errors.New("RowIDs() argument required: field")
+	}
+	// Fetch field.
+	f := e.Holder.Field(index, fieldName)
+	if f == nil {
+		return nil, ErrFieldNotFound
+	}
+	frag := e.Holder.fragment(index, fieldName, viewStandard, shard)
+	if frag == nil {
+		return make(RowIDs, 0), nil
+	}
+	if columnID, ok, err := c.UintArg("column"); err != nil {
+		return nil, err
+	} else if ok {
+		// TODO: it's possible that filters could be applied here, so this returns too early.
+		return frag.rowsForColumn(columnID), nil
+	}
+	filter := getFilterFunction(c)
+	return frag.rowsWithFilter(filter), nil
+}
+
+// getGroupByFilterFunction returns a rowFilter based on the
+// field directive provided.
+func getGroupByFilterFunction(fieldDirective string) (rowFilter, error) {
+	parts := strings.Split(fieldDirective, ":")
+	hasLimit := false
+	hasOffset := false
+	limit := uint64(0)
+	offset := uint64(0)
+	var err error
+	// fieldDirective can have one of the following forms:
```

Comment:
this comment should be part of the doc string on the function IMO - even though it isn't exported